src/main/java/org/java_lcw/jil/AWTImage.java

```diff
@@ -0,0 +1,485 @@
+    } else {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+    }
+    nbi.getGraphics().drawImage(bi, 0, 0, null);
+    return new AWTImage(nbi, MODE);
+  }
+
+  @Override
+  public AWTImage resizeWithBorders(int bWidth, int bHeight, Color borderColor,
+      ScaleType st) {
+    return null;
+  }
+
+  @Override
+  public AWTImage resize(int newWidth, int newHeight, boolean keepAspect, ScaleType st) {
+    if(keepAspect) {
+      int[] aspect = Utils.getAspectSize(this.width, this.height, newWidth, newHeight);
+      newWidth = aspect[0];
+      newHeight = aspect[1];
+    }
+    AWTImage nimg;
+    BufferedImage resizedImage = new BufferedImage(newWidth, newHeight, bi.getType());
+    Graphics2D g = resizedImage.createGraphics();
+
+    switch(st) {
+    case LINER:{
+      g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
+      g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_OFF);
+      g.drawImage(bi, 0, 0, newWidth, newHeight, null);
+      g.dispose();
+      nimg = new AWTImage(resizedImage, bpp);
+    } break;      
+    case CUBIC:{
+      g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BICUBIC);
+      g.drawImage(bi, 0, 0, newWidth, newHeight, null);
+      g.dispose();
+      nimg = new AWTImage(resizedImage, bpp);
+    } break;
+    case CUBIC_SMOOTH:
+      nimg = (AWTImage)Utils.biCubicSmooth(this, newWidth, newHeight);
+    default:{
+      g.setRenderingHint(RenderingHints.KEY_COLOR_RENDERING, RenderingHints.VALUE_COLOR_RENDER_SPEED);
+      g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR);
+      g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_OFF);
+      g.drawImage(bi, 0, 0, newWidth, newHeight, null);
+      g.dispose();
+      nimg = new AWTImage(resizedImage, bpp);
+    }break;
+    }
+
+    return nimg;
+  }
+
+  @Override
+  public void fillImageWithColor(Color c) {
+    Graphics2D graphics = bi.createGraphics();
+    graphics.setPaint(Utils.toAWTColor(c));
+    graphics.fillRect(0, 0, width, height);
+    graphics.dispose();
+  }
+
+  @Override
+  public void setPixel(int x, int y, Color c) {
+    bi.setRGB(x, y, Utils.toAWTColor(c).getRGB());
+  }
+
+  @Override
+  public Color getPixel(int x, int y) {
+    return Utils.toJILColor(new java.awt.Color(bi.getRGB(x, y)));
+  }
+
+  @Override
+  public void paste(int x, int y, Image img) {
+    BufferedImage bi2;
+    if(img instanceof AWTImage) {
+      bi2 = ((AWTImage)img).bi;
+    } else {
+      bi2 = AWTImage.fromJavaImage(img.toJavaImage()).bi;
+    }
+    Graphics2D gi = bi.createGraphics();
+    float opacity = 1.0f;
+    gi.setComposite(AlphaComposite.getInstance(AlphaComposite.SRC, opacity));
+    gi.drawImage(bi2, x, y, null);
+    gi.dispose();
```

Comment:
I see there are other cases of this...same as above. (but I will stop commenting on them, so may need to search for dispose instances)